server/plugin/virtual_agent_test.go

```diff
@@ -212,3 +219,102 @@ func Test_CreatePickerAttachment(t *testing.T) {
 		})
 	}
 }
+
+func Test_CreateOutputImagePost(t *testing.T) {
+	defer monkey.UnpatchAll()
+
+	for _, testCase := range []struct {
+		description           string
+		body                  *OutputImage
+		getDirectChannelError *model.AppError
+		uploadFileError       *model.AppError
+		isErrorExpected       bool
+		expectedError         string
+		readAllError          error
+		httpGetError          error
+		contentType           string
+	}{
+		{
+			description: "Image post is created",
+			body: &OutputImage{
+				Value:   "https://test/test.jpg",
+				AltText: "mockAltText",
+			},
+			contentType: "image/jpg",
+		},
+		{
+			description: "No image post is created due to invalid image URL",
+			body: &OutputImage{
+				Value:   "htps://test/test.jpg",
+				AltText: "mockAltText",
+			},
+			isErrorExpected: true,
+			httpGetError:    errors.New("unsupported protocol scheme"),
+			expectedError:   "unsupported protocol scheme",
+		},
+		{
+			description: "Not able to get direct channel",
+			body: &OutputImage{
+				Value:   "https://test/test.jpg",
+				AltText: "mockAltText",
+			},
+			getDirectChannelError: &model.AppError{
+				Message: "mockErrorMessage",
+			},
+			isErrorExpected: true,
+			expectedError:   "mockErrorMessage",
+		},
+		{
+			description: "Not able to upload file on Mattermost",
+			body: &OutputImage{
+				Value:   "https://test/test.jpg",
+				AltText: "mockAltText",
+			},
+			uploadFileError: &model.AppError{},
+			contentType:     "image/jpg",
+		},
+		{
+			description: "Error reading file data",
+			body: &OutputImage{
+				Value:   "https://test/test.jpg",
+				AltText: "mockAltText",
```

Comment:
This is same in every test case. Define this outside the loop and use that variable here.